### ISSUE TITLE
Grad seems not be collected between gpus before updating in vit example code

### ISSUE THREAD

**Issue by frankxyy** [alpa-projects/alpa#835]
Relevant code:

https://github.com/alpa-projects/alpa/blob/f2fd1488a9a075e135c06a63e1c745d48c01be65/examples/ViT/run_image_classification.py#L454

**Comment by merrymercy** [comment]
I think alpa will automatically insert the necessary communication to gather the gradient.
We mark the gradient by https://github.com/alpa-projects/alpa/blob/f2fd1488a9a075e135c06a63e1c745d48c01be65/examples/ViT/run_image_classification.py#L452

**Comment by frankxyy** [comment]
@merrymercy I found that jax is async dispatched: https://jax.readthedocs.io/en/latest/async_dispatch.html#async-dispatch. I also found in the flax example code there is calling of `grads = lax.pmean(grads, axis_name='batch')` which I assume it to actually activate the computing.
![image](https://user-images.githubusercontent.com/6971044/210313098-b1e4062e-34d9-4eec-a130-93cade8e8a2d.png)

I am not really sure about my observasion and willing to have discussion with you.
